### PR TITLE
AWX Collections for DAB RBAC

### PR DESCRIPTION
**Pull request by fosterseth**
##### SUMMARY

Adds new modules for CRUD operations on the following endpoints:

- api/v2/role_definitions
- api/v2/role_user_assignments
- api/v2/role_team_assignments

Note: assignment is Create or Delete only

Additional changes:
- Currently DAB endpoints do not have "type" field on the resource list items. So this modifies the create_or_update_if_needed to allow manually specifying item type.

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Collection

### TODO

- [ ] We discussed adding "type" field to DAB endpoint items. This PR does not do that, and the work around is to just specify the type manually where needed

### CODE REVIEW

File: awx_collection/plugins/modules/role_definition.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    - Contains a list of permissions and a resource type that can then be assigned to users or teams.
+options:
+    name:
+        description:
+            - Name of this role definition.
+        required: True
+        type: str
+    permissions:
+        description:
+            - List of permission ids to include in the role definition.
```
Comment (fosterseth) [inline awx_collection/plugins/modules/role_definition.py]:
```suggestion
            - List of permissions to include in the role definition.
```

File: awx_collection/test/awx/test_role_definition.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    role_definition = RoleDefinition.objects.get(name='test_launch_job')
+    permission_codenames = [p.codename for p in role_definition.permissions.all()]
+    assert set(permission_codenames) == set(['view_jobtemplate'])
+    assert role_definition.content_type.model == 'jobtemplate'
+
+
+    result = run_module(
+        'role_definition',
+        {
+            'name': 'test_launch_job',
```
Comment (AlanCoding) [inline awx_collection/test/awx/test_role_definition.py:51]: name not meaningful, needs name like 'test_view_jt'

File: awx_collection/test/awx/test_role_definition.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    assert set(permission_codenames) == set(['view_jobtemplate', 'execute_jobtemplate'])
+    assert role_definition.content_type.model == 'jobtemplate'
+
+
+@pytest.mark.django_db
+def test_update_existing(run_module, admin_user):
+    result = run_module(
+        'role_definition',
+        {
+            'name': 'test_launch_job',
```
Comment (AlanCoding) [inline awx_collection/test/awx/test_role_definition.py:34]: name not meaningful, needs name like 'test_view_jt'

**Comment by AlanCoding** [comment]
collection sanity still failing on

```
2024-04-02T19:48:54.0150623Z ERROR: test/awx/test_role_definition.py:48:5: E303: too many blank lines (2)
```

EDIT: didn't mean to click close button.

**Review by AlanCoding — APPROVED** [review]
This looks great, thank you for doing it! This is a big help to get things formally complete.

**Pull request merged**